owncloudApp/src/main/java/com/owncloud/android/presentation/sharing/shares/PublicShareDialogFragment.kt

```diff
@@ -321,6 +343,111 @@ class PublicShareDialogFragment : DialogFragment() {
         })
     }
 
+    private fun requirementsPasswordPolicy(password: String, passwordPolicy: OCCapability.PasswordPolicy) {
+
+            var hasMaxCharacters = password.length <= passwordPolicy.maxCharacters!!
+            var hasMinCharacters = password.length >= passwordPolicy.minCharacters!!
+            var hasUpperCase = password.count { it.isUpperCase() } >= passwordPolicy.minUppercaseCharacters!!
+            var hasLowerCase = password.count { it.isLowerCase() } >= passwordPolicy.minLowercaseCharacters!!
+            var hasSpecialCharacter = password.count { SPECIALS_CHARACTERS.contains(it) } >= passwordPolicy.minSpecialCharacters!!
+            var hasDigit = password.count { it.isDigit() } >= passwordPolicy.minDigits!!
```

Comment:
Shouldn't be the case, but what about if any of these policies is `null`? Let's make safer code by handling nullability

Reply:
Done, logic improved.